Index: lib/IR/AsmWriter.cpp
===================================================================
--- lib/IR/AsmWriter.cpp	(revision 227257)
+++ lib/IR/AsmWriter.cpp	(working copy)
@@ -2030,13 +2030,14 @@
     Out << "\n";
     PrintLLVMName(Out, BB->getName(), LabelPrefix);
     Out << ':';
-  } else if (!BB->use_empty()) {      // Don't print block # of no uses...
-    Out << "\n; <label>:";
+  } else {      // Print block # for unnamed basic blocks.
+    Out << "\n";
     int Slot = Machine.getLocalSlot(BB);
     if (Slot != -1)
       Out << Slot;
     else
       Out << "<badref>";
+    Out << ':';
   }
 
   if (!BB->getParent()) {
